packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -1430,10 +1430,11 @@ class FlutterPlugin implements Plugin<Project> {
                         }
                     }
                 }
-                // Copy the native assets created by build.dart and placed here by flutter assemble.
-                String nativeAssetsDir = "${project.buildDir}/../native_assets/android/jniLibs/lib/"
-                project.android.sourceSets.main.jniLibs.srcDir(nativeAssetsDir)
             }
+            // Copy the native assets created by build.dart and placed here by flutter assemble.
+            // This path is not flavor specific and must only be added once.
+            String nativeAssetsDir = "${project.buildDir}/../native_assets/android/jniLibs/lib/"
```

Comment:
Add a reference to https://github.com/dart-lang/native/issues/1359, in case we add support for flavors to native assets. (And then we need to figure out how to only add it once per flavor and not duplicate it from all flavors.)